classes/use-imx-seco-firmware.bbclass

```diff
@@ -0,0 +1,32 @@
+#
+# Class to provide a possibility to re-use names of SECO Firmware
+# files, which are required by certain derivatives on i.MX8 family.
+#
+# SECO Firmware names are used by imx-boot and imx-seco recipes,
+# therefore their respective definitions are moved into a separate
+# class, which is inherited by both recipes.
+#
+# Currently, only 'mx8x' family requires SECO FW to be provided in the
+# target image, therefore as a first step - the machine override is
+# verified if it matches, and then derivative is taken.
+#
+# NOTE: SECO_FIRMWARE_NAME defaults to empty string, and are verified
+# against the family first. If a derivative in the family does not
+# have a firmware name set in this class - recipe parsing is stopped.
+# This behavior ensures that derivatives which requires SECO Firmware
+# to be present in the image file have it properly defined.
+
+SECO_FIRMWARE_NAME              ?= ""
+
+SECO_FIRMWARE_NAME_mx8qm         = "mx8qmb0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8qxp        = "mx8qxb0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8qxpc0      = "mx8qxc0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8phantomdxl = "mx8qxb0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8dxl        = "mx8dxla0-ahab-container.img"
+
+python () {
+    if "mx8x" in d.getVar('MACHINEOVERRIDES').split(":"):
+        seco_firmware = d.getVar('SECO_FIRMWARE_NAME')
+        if not seco_firmware:
+            raise bb.parse.SkipRecipe("This SoC requires 'SECO_FIRMWARE_NAME', define it in 'use-imx-seco-firmware' bbclass")
+}
```

Comment:
I believe we don't need to check the `mx8x` on `MACHINEOVERRIDES` as it'll fail if we don't have the firmware on the list. So it is redundant.

Reply:
I already made a suggestion above based on comment from @thochstein that filter would instead be looking at all overrides but `mx8m` where SECO FW is not required. Without that in place `imx-boot` fails to build for `mx8m` derivatives since the `SECO_FIRMWARE_NAME` is set to empty string.

Would it be OK to adjust the parsing filter like I suggested above?